analyzers/maven/maven.go

```diff
@@ -130,64 +132,39 @@ func (a *Analyzer) IsBuilt() (bool, error) {
 func (a *Analyzer) Analyze() (graph.Deps, error) {
 	log.WithField("module", a.Module).Debug("analyzing module")
 
-	var imports []maven.Dependency
-	var deps map[maven.Dependency][]maven.Dependency
-	var err error
-	if a.Options.Command == "" {
-		imports, deps, err = a.Maven.DependencyTree(a.Module.Dir, a.Module.BuildTarget)
-	} else {
-		var output string
-		output, _, err = exec.Shell(exec.Cmd{
-			Command: a.Options.Command,
-		})
+	switch a.Options.Strategy {
+	case "pom-file":
+		pom, err := a.Maven.ResolveManifestFromBuildTarget(a.Module.BuildTarget)
 		if err != nil {
 			return graph.Deps{}, err
 		}
-		imports, deps, err = maven.ParseDependencyTree(output)
-	}
-	if err != nil {
-		return graph.Deps{}, err
-	}
-
-	// Set direct dependencies.
-	var i []pkg.Import
-	for _, dep := range imports {
-		i = append(i, pkg.Import{
-			Resolved: pkg.ID{
-				Type:     pkg.Maven,
-				Name:     dep.Name,
-				Revision: dep.Version,
-			},
-		})
-	}
-
-	// Set transitive dependencies.
-	g := make(map[pkg.ID]pkg.Package)
-	for parent, children := range deps {
-		id := pkg.ID{
-			Type:     pkg.Maven,
-			Name:     parent.Name,
-			Revision: parent.Version,
-		}
-		var imports []pkg.Import
-		for _, child := range children {
-			imports = append(imports, pkg.Import{
-				Target: child.Version,
-				Resolved: pkg.ID{
-					Type:     pkg.Maven,
-					Name:     child.Name,
-					Revision: child.Version,
-				},
+		return pom.ToGraphDeps(), nil
+	case "maven-tree":
+		return a.Maven.DependencyTree(a.Module.Dir, a.Module.BuildTarget)
+	default:
+		if a.Options.Command != "" {
+			output, _, err := exec.Shell(exec.Cmd{
+				Command: a.Options.Command,
 			})
+			if err != nil {
+				// Because this was a custom shell command, we do not fall back to any other strategies.
+				return graph.Deps{}, err
+			}
+			return maven.ParseDependencyTree(output)
 		}
-		g[id] = pkg.Package{
-			ID:      id,
-			Imports: imports,
+		deps, err := a.Maven.DependencyTree(a.Module.Dir, a.Module.BuildTarget)
+		if err != nil || len(deps.Direct) == 0 {
+			log.Warnf(
+				"Could not use Maven to determine dependencies for %q. Falling back to use manifest file.",
+				a.Module.Name)
+			mvnError := err
+			pom, err := a.Maven.ResolveManifestFromBuildTarget(a.Module.BuildTarget)
+			if err != nil {
+				return graph.Deps{},
+					errors.Wrapf(err, "could not identify dependencies; original error: %v", mvnError)
+			}
+			return pom.ToGraphDeps(), nil
```

Comment:
we need something like `ResolveManifestFromBuildTarget` to get just the contents of a POM file because we sometimes need things like the `groupId`, `artifactId`, and `parent` from it